### PR TITLE
feat: commplet 16 programs

### PR DESCRIPTION
**Pull request by nived-cv**
MR for completed 16 programs

### CODE REVIEW

File: schoolManagement.js [hunk 1]
```diff
@@ -1 +1,261 @@
...
+}
+
+const printStudentIds = ()=>{
+    classObj.students.forEach((student)=>{
+        console.log(student.id)
+    })
+}
+
+const printSubjectsOfOne = (name)=>{
+    let {students} = classObj
```
Comment (Ansam-C-D) [inline schoolManagement.js:74]: we can use find method in javascript to take only one student
something like this

   ```
 let student = classObj.students.find(student => student.id === studentId)
   student?.marks.forEach((subjectDetails) =>
    console.log(subjectDetails.subject)
  )
```
  also what if there is no matching in student? so we should print an exception if no students found
  like
   ` if(!student){console.log('student not found')}`

File: schoolManagement.js [hunk 2]
```diff
@@ -1 +1,261 @@
...
+    
+    students.forEach((student)=> {
+        if(student.name == name) 
+            student.marks.forEach(subject => console.log(subject.subject))
+    })
+}
+
+const printMarksOfOne = (name)=>{
```
Comment (Ansam-C-D) [inline schoolManagement.js:82]: is findStudentMark a better name

File: schoolManagement.js [hunk 3]
```diff
@@ -1 +1,261 @@
...
+    })
+}
+
+const printStudentIds = ()=>{
+    classObj.students.forEach((student)=>{
+        console.log(student.id)
+    })
+}
+
+const printSubjectsOfOne = (name)=>{
```
Comment (Ansam-C-D) [inline schoolManagement.js:73]: findStudentSubjects

File: schoolManagement.js [hunk 4]
```diff
@@ -1 +1,261 @@
...
+        {
+            total=student.marks.reduce((total,value)=>total+=value.mark,0)
+            number = student.marks.length
+            average = total/number
+        }
+    })
+    console.log(average)
+
+}
+const totalMarkOfOne =(name)=>{
```
Comment (Ansam-C-D) [inline schoolManagement.js:104]: leave a line above

File: schoolManagement.js [hunk 5]
```diff
@@ -1 +1,261 @@
...
+        if(student.name == name) 
+            student.marks.forEach(subject => console.log(subject.subject))
+    })
+}
+
+const printMarksOfOne = (name)=>{
+    let {students} = classObj
+    
+    students.forEach((student)=> {
+        if(student.name == name)
```
Comment (Ansam-C-D) [inline schoolManagement.js:86]: here also we can use the method that I mentioned above. also change name to id for checking

File: schoolManagement.js [hunk 6]
```diff
@@ -1 +1,261 @@
...
+    let {students} = classObj
+    
+    students.forEach((student)=> {
+        if(student.name == name) 
+            student.marks.forEach(subject => console.log(subject.subject,subject.mark))
+    })
+}
+
+const averageMarkOfOne =(name)=>{
+    let {students} = classObj
```
Comment (Ansam-C-D) [inline schoolManagement.js:92]: there are so many formatting issues.
leave a space inside curly braces
`let { students } = classObj`

File: schoolManagement.js [hunk 7]
```diff
@@ -1 +1,261 @@
...
+    students.forEach((student)=>{
+        if(student.name == name)
+        {
+            total=student.marks.reduce((total,value)=>total+=value.mark,0)
```
Comment (Ansam-C-D) [inline schoolManagement.js:96]: always leave a space after variables and operators to look good

 total = student.marks.reduce((total, value) => (total += value.mark), 0)
same for all functions

File: schoolManagement.js [hunk 8]
```diff
@@ -1 +1,261 @@
...
+    console.log(average)
+
+}
+const totalMarkOfOne =(name)=>{
+    let {students} = classObj
+    students.forEach((student)=>{
+        if(student.name == name)
+        total=student.marks.reduce((total,value)=>total+=value.mark,0)
+    })
+    console.log(total)
```
Comment (Ansam-C-D) [inline schoolManagement.js:110]: leave a line above

File: schoolManagement.js [hunk 9]
```diff
@@ -1 +1,261 @@
...
+}
+
+const averageMarkOfSubject = (sub) =>{
+
```
Comment (Ansam-C-D) [inline schoolManagement.js:114]: remove line here

**Review by Ansam-C-D — CHANGES_REQUESTED** [review]

File: schoolManagement.js [hunk 10]
```diff
@@ -1 +1,261 @@
...
+const printClassName = ()=>{
+    console.log(classObj.name)
+}
+
+const printTeacherName = ()=>{
+    console.log(classObj.teacherName)
+}
+
+const printStudentNames = ()=>{
```
[the remark this answers is not in the document]
Comment (nived-cv) [reply schoolManagement.js:61]: changed

File: schoolManagement.js [hunk 11]
```diff
@@ -1 +1,261 @@
...
+        }
+    ]
+}
+// code
+
+const printClassName = ()=>{
+    console.log(classObj.name)
+}
+
+const printTeacherName = ()=>{
```
[the remark this answers is not in the document]
Comment (nived-cv) [reply schoolManagement.js:57]: printing only one teacher

**Pull request merged**